internal/util/testing/context.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package util provides utilities for net.Listener.
+package util
+
+import (
+	"context"
+	"testing"
+)
+
+type contextTestKey string
+
+// NewContextForUnitTest returns a context appropriate for a service test that is not in test/e2e/.
+func NewContextForUnitTest(t *testing.T) context.Context {
```

Comment:
Naming Suggestion: TestingContext().

Also, maybe with a shorter name you can do:
```
			ctx, cancel := context.WithCancel(utilTesting.TestingContext())
```

Instead of:
```
			ctx := utilTesting.NewContextForUnitTest(t)
			ctx, cancel := context.WithCancel(ctx)
```
In a few places.

Reply:
Renamed to NewContext(). In golang they discourage "stuttering" names. So `utilTesting.NewContext()` seems like the best here.